x-pack/metricbeat/module/aws/ec2/ec2_test.go

```diff
@@ -2,554 +2,20 @@
-			0.2,
-			1.0,
-		},
-	}
-
-	calculateRate(resultMetricsetFields, "disabled")
-	for _, c := range cases {
-		output, err := resultMetricsetFields.GetValue(c.rateMetricName)
-		assert.NoError(t, err)
-		assert.Equal(t, c.rateMetricValue, output)
-	}
-
-	calculateRate(resultMetricsetFields, "enabled")
-	for _, c := range cases {
-		output, err := resultMetricsetFields.GetValue(c.rateMetricName)
-		assert.NoError(t, err)
-		assert.Equal(t, c.rateMetricValueDetailed, output)
-	}
-}
-
-func TestCreateCloudWatchEventsWithInstanceName(t *testing.T) {
-	expectedEvent := mb.Event{
-		RootFields: common.MapStr{
-			"cloud": common.MapStr{
-				"region":            regionName,
-				"provider":          "aws",
-				"instance":          common.MapStr{"id": "i-123", "name": "test-instance"},
-				"machine":           common.MapStr{"type": "t2.medium"},
-				"availability_zone": "us-west-1a",
-			},
-			"host": common.MapStr{
-				"cpu": common.MapStr{"usage": 0.25},
-				"id":  "i-123",
-			},
-		},
-		MetricSetFields: common.MapStr{
-			"tags": common.MapStr{
-				"app_kubernetes_io/name": "foo",
-				"helm_sh/chart":          "foo-chart",
-				"Name":                   "test-instance",
-			},
-		},
-	}
-	svcEC2Mock := &MockEC2Client{}
-	instanceIDs, instancesOutputs, err := getInstancesPerRegion(svcEC2Mock)
-	assert.NoError(t, err)
-	assert.Equal(t, 1, len(instanceIDs))
-	instanceID := instanceIDs[0]
-	assert.Equal(t, instanceID, instanceID)
-	timestamp := time.Now()
-
-	getMetricDataOutput := []cloudwatch.MetricDataResult{
-		{
-			Id:         &id1,
-			Label:      &label1,
-			Values:     []float64{0.25},
-			Timestamps: []time.Time{timestamp},
-		},
-	}
-
-	metricSet := MetricSet{
-		&aws.MetricSet{},
-		logp.NewLogger("test"),
-	}
-
-	events, err := metricSet.createCloudWatchEvents(getMetricDataOutput, instancesOutputs, "us-west-1")
-	assert.NoError(t, err)
-	assert.Equal(t, 1, len(events))
-
-	assert.Equal(t, expectedEvent.MetricSetFields["tags"], events[instanceID].ModuleFields["tags"])
-
-	hostID, err := events[instanceID].RootFields.GetValue("host.id")
-	assert.NoError(t, err)
-	assert.Equal(t, "i-123", hostID)
-
-	instanceName, err := events[instanceID].RootFields.GetValue("cloud.instance.name")
-	assert.NoError(t, err)
-	assert.Equal(t, "test-instance", instanceName)
-}
-
-func TestNewLabel(t *testing.T) {
-	instanceID := "i-123"
-	metricName := "CPUUtilization"
-	statistic := "Average"
-	label := newLabel(instanceID, metricName, statistic).JSON()
-	assert.Equal(t, "{\"InstanceID\":\"i-123\",\"MetricName\":\"CPUUtilization\",\"Statistic\":\"Average\"}", label)
-}
-
-func TestConvertLabel(t *testing.T) {
-	labelStr := "{\"InstanceID\":\"i-123\",\"MetricName\":\"CPUUtilization\",\"Statistic\":\"Average\"}"
-	label, err := newLabelFromJSON(labelStr)
-	assert.NoError(t, err)
-	assert.Equal(t, "i-123", label.InstanceID)
-	assert.Equal(t, "CPUUtilization", label.MetricName)
-	assert.Equal(t, "Average", label.Statistic)
+func init() {
```

Comment:
Yeah this file is just usef for the init(). It's set to be like this when we started using light weight module but I'm sure there is a better spot for this init function now. Is it OK if I keep it like this for now? I need to create a separate PR to change this in all metricsets under aws module since all the other metricsets are using this convention right now.

The tests are removed because we don't have those functions anymore. They are under cloudwatch metricset and being tested there already :)

Reply:
Ok I see, no problem to leave it as is then! Thanks for clarifying!